support/mono_embeddinator.c

```diff
@@ -314,3 +314,67 @@ MonoClass* mono_embeddinator_get_decimal_class ()
     }
     return decimalclass;
 }
+
+MonoClass* mono_embeddinator_get_datetime_class ()
+{
+    static MonoClass* datetimeclass = nil;
+    if (!datetimeclass) {
+        datetimeclass = mono_class_from_name (mono_get_corlib (), "System", "DateTime");
+    }
+    return datetimeclass;
+}
+
+int mono_embeddinator_get_datetime_size ()
+{
+    static int datetimesize = 0;
+    if (datetimesize == 0) {
+        datetimesize = mono_class_instance_size (mono_embeddinator_get_datetime_class ());
+    }
+    return datetimesize;
+}
+
+E4KDateTimeDefaults* mono_embeddinator_get_datetime_defaults (mono_embeddinator_context_t* context)
+{
+	static E4KDateTimeDefaults datetimedefaults;
+	static bool initialized = false;
+	if (!initialized) {
+		MonoClass* timespanclass = mono_class_from_name (mono_get_corlib (), "System", "TimeSpan");
+		MonoClassField* tickspersecondfield = mono_class_get_field_from_name (timespanclass, "TicksPerSecond");
+		MonoVTable* timespanvtable = mono_class_vtable (context->domain, timespanclass);
+		mono_field_static_get_value (timespanvtable, tickspersecondfield, &datetimedefaults.NetTicksPerSecond);
+
+		MonoClass* datetimeclass = mono_embeddinator_get_datetime_class ();
+		MonoVTable* datetimevtable = mono_class_vtable (context->domain, datetimeclass);
+		mono_runtime_class_init (datetimevtable);
+		MonoClassField* dtmaxvaluefield = mono_class_get_field_from_name (datetimeclass, "MaxValue");
+		MonoClassField* dtminvaluefield = mono_class_get_field_from_name (datetimeclass, "MinValue");
+
+		int datetimesize = mono_embeddinator_get_datetime_size ();
+		void* dtmaxvalueptr = malloc (datetimesize);
+		void* dtminvalueptr = malloc (datetimesize);
+		mono_field_static_get_value (datetimevtable, dtmaxvaluefield, dtmaxvalueptr);
+		mono_field_static_get_value (datetimevtable, dtminvaluefield, dtminvalueptr);
+
+		MonoMethod* dtticks = mono_embeddinator_lookup_method ("System.DateTime:get_Ticks()", datetimeclass);
+
+		MonoObject* dtmaxtickserr = nil;
+		MonoObject* dtmaxticksboxed = mono_runtime_invoke (dtticks, dtmaxvalueptr, NULL, &dtmaxtickserr);
+		if (dtmaxtickserr)
+			mono_embeddinator_throw_exception (dtmaxtickserr);
+
+		MonoObject* dtmintickserr = nil;
+		MonoObject* dtminticksboxed = mono_runtime_invoke (dtticks, dtminvalueptr, NULL, &dtmintickserr);
+		if (dtmintickserr)
+			mono_embeddinator_throw_exception (dtmintickserr);
+
+		datetimedefaults.MaxValueTicks = *(long long *) mono_object_unbox (dtmaxticksboxed);
+		datetimedefaults.MinValueTicks = *(long long *) mono_object_unbox (dtminticksboxed);
```

Comment:
Yeah, I was trying to play safe by actually get the values from the runtime but I guess I was just over-engeneering this so if no objections I will harcode the values.